pkg/auth/poptoken/rsakeymanager.go

```diff
@@ -0,0 +1,80 @@
+package poptoken
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"sync"
+	"time"
+)
+
+type RsaKeyPair struct {
+	PrivateKey *rsa.PrivateKey
+	PublicKey  *rsa.PublicKey
+	RsaSize    int
+	Kty        string
+	Alg        string
+}
+
+// a RSA Key generator that refresh the RSA KeyPair at regular interval
+// Used to ensure the keys use to sign the poptoken are rotated
+type rsaKeyManager struct {
+	refreshInterval time.Duration
+	createdDateTime time.Time
+	privateKey      *rsa.PrivateKey
+	mutex           sync.Mutex
+}
+
+const (
+	DefaultRefreshInterval = time.Hour * 8
+	RsaSize                = 2048
+	Kty                    = "RSA"
+	Alg                    = "RS256"
+)
+
+func generatePrivateKey() (*rsa.PrivateKey, error) {
+	return rsa.GenerateKey(rand.Reader, RsaSize)
+}
+
+// Return a KeyPair. The keypair is its own copy and not a reference.
+func (r *rsaKeyManager) GetKeyPair(now time.Time) (*RsaKeyPair, error) {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
```

Comment:
I know my sample I shared had this, but thinking on it more we can drop this - it should be fine if we get multiple concurrent calls into this, as each returned instance gets a valid key for its lifetime and at worst we just have a couple rotations that could occur with last write wins